New "Shows" Page

alex-ruddell opened the pull request:
Replacement page for "previous" working. Not yet linked with website, but can access directly to check if it is working properly. 
Hopefully can get rid of the previous page if all goes well.

alex-kennedy commented on src/components/shows/heroBanner.js:
```diff
@@ -0,0 +1,80 @@
+      // If a year HAS been entered...
+      return (
+        <section
+          style={{
```
Maybe this would be tidier in the CSS files, setting a class or ID for this section.

alex-ruddell replied:
Done! :)

alex-kennedy commented on src/components/shows/heroBanner.module.css:
```diff
@@ -0,0 +1,75 @@
+.hero {
```
If you can be arsed, CSS styles for each class/ID should be ordered alphabetically.

alex-ruddell replied:
Also done :)

alex-kennedy commented on src/components/shows/data.js:
```diff
@@ -0,0 +1,331 @@
+import revue2019 from "../../assets/images/previous/2019.jpg"
```
Maybe it's worth asking someone else to proofread this as well?

alex-ruddell replied:
Definitely, I have had another read through myself for the time being :D

dikunong replied:
A few missing apostrophes which I commented on below but otherwise all looks great!

alex-kennedy commented:
Oh only other thing is that the shows page isn't accessible yet, should we consider removing the old previous page?

alex-ruddell commented on src/components/shows/showviewer.js:
```diff
@@ -0,0 +1,194 @@
+    //                      Small fix done but not perfect.
+    // =====================================================
+
+    var next = parseInt(this.state.year) + 1;
```
Thank you!

dikunong commented on src/components/header.js:
```diff
@@ -26,7 +26,7 @@ class Header extends React.Component {
         <div className={styles.logoArea}>
           <h3 style={{ margin: 0 }}>
-            <Link to="/">ENGINEERING REVUE</Link>
+            <Link to="/">AUCKLAND ENGINEERING REVUE</Link>
```
This should stay as "ENGINEERING REVUE", otherwise it's too long and doesn't fit on mobile.

![image](https://user-images.githubusercontent.com/1442037/79942280-982b9d00-84ba-11ea-8d23-1678b504c818.png)

```suggestion
            <Link to="/">ENGINEERING REVUE</Link>
```

dikunong commented on src/components/shows/data.js:
```diff
@@ -0,0 +1,331 @@
+    "director": "Janet Van",
+    "producer": "Luke Harries",
+    "blurb": `In the fifth annual Auckland Engineering Revue, the 100% University-of-Auckland based cast attacks a wonderful animated
+                childrens film in the hopes of not ruining it for their mature audience! In this spin-off parody of "Finding Nemo", you 
```
Found one heh

dikunong commented on src/components/shows/data.js:
```diff
@@ -0,0 +1,331 @@
+    "director": "Mahmood Hikmet & Nick Corlett",
+    "producer": "Emma Foulkes",
+    "blurb": `In the fourth annual Auckland Engineering Revue, the 100% University-of-Auckland based cast attacks a wonderful animated
+                childrens film in the hopes of not ruining it for their mature audience! In this spin-off parody of "The Lion King", you 
```
Here too heh

dikunong commented on src/components/shows/heroBanner.module.css:
```diff
@@ -0,0 +1,75 @@
+  }
+}
+
+@media only screen and (max-width: 425px) {
```
These font sizes are too small. Text should never really get smaller than the body size (16px) unless in exceptional circumstances as it's just too difficult to read.
In general, your heading and body text sizes should be more like the second image. Don't be afraid to use more vertical space, smartphones are tall.

![image](https://user-images.githubusercontent.com/1442037/79943487-1c7f1f80-84bd-11ea-9246-415691248058.png)

dikunong commented on src/components/shows/showviewer.module.css:
```diff
@@ -0,0 +1,189 @@
+
+.view {
+  max-height: 100%;
+}
```
This CSS needs media queries to handle resizing for smaller screens (think mainly phones, but also tablets). A few points:

* At many widths, an awkward gap is created when the screen can't quite fit another column of posters. To resolve this, centre the posters so the gap is distributed either side.
![image](https://user-images.githubusercontent.com/1442037/79943707-b646cc80-84bd-11ea-8962-a0787fee457f.png)

* At single-poster widths, the touch boxes of the posters should ideally fill the entire screen width. This is to avoid users tapping on dead space on mobile.
![image](https://user-images.githubusercontent.com/1442037/79943837-1473af80-84be-11ea-97d1-901aad7ff3d2.png)

* The next and previous show buttons get really awkward when squished into a narrow width. Unsure what the best solution is - shrinking the text would get unreadable really quick, hmm.
![image](https://user-images.githubusercontent.com/1442037/79943885-408f3080-84be-11ea-8584-8e2206de018f.png)

* Percentage layouts break down at smaller widths. Beyond a certain point, you're best off switching to a one-column layout (so it would go quote - poster picture - about the show - the cast - cast picture - sign up). This can be done using a media query to change the flexbox properties.
![image](https://user-images.githubusercontent.com/1442037/79944024-a4b1f480-84be-11ea-9a73-6b2573934a30.png)

* The "Click here to sign up!" button breaks apart when it doesn't have enough width.
![image](https://user-images.githubusercontent.com/1442037/79944110-d88d1a00-84be-11ea-93f7-47be52db13c0.png)

Most of this was discovered by using Chrome Dev tools to test a phone layout, which you can access by pressing F12 and clicking the second icon from the top-left ("Toggle device toolbar" when hovered). I usually test the iPhone X size first, but ideally your website should respond fluidly as you resize it (set to Responsive and then drag the handle on the right hand side of the "window") to ensure it looks great on any phone/tablet/desktop window size! (The smallest width you need to test is 320px, which matches the 2016 and older iPhones).

alex-ruddell replied:
Fixed all bar poster touch boxes, which I don't particularly see to be a problem?

dikunong commented on src/components/shows/showviewer.js:
```diff
@@ -0,0 +1,194 @@
+              {/* To display specific show information*/}
+              <div className={styles.topButtons}>
+                
+                <div className={styles.fwrdButton}>
```
These buttons are not accessible to people who cannot use a mouse. These should be wrapped in a `<button>` element so that the browser will take care of this for you. (This also brings additional benefits like applying the proper "hand" cursor without needing to set it in CSS).

dikunong commented on src/components/shows/showviewer.js:
```diff
@@ -0,0 +1,194 @@
+import React from "react"
```
Some general design things to think about:

* When viewing a show, there is no back/close button, only next and previous show buttons, on all but the most and least recent shows. A back/close button should always be present in the UI somewhere when viewing a "detail" screen so that the user doesn't feel trapped or stuck.
![image](https://user-images.githubusercontent.com/1442037/79944801-906ef700-84c0-11ea-9c2c-af7107093155.png)

* Extending from the previous point, you might want to try and store the user's viewed shows in the browser's Back history so that the browser Back button can go back. That would probably be really hard though so worry about it later.

* The main viewer makes heavy use of hover-over text elements. These are cool, but less accessible and notably people on phones/touchscreen devices can't see them at all. This doesn't affect usability too much (the posters have `alt` text which blind people can access) but is worth thinking about.

* The hover-over transition speed is also a bit slow at 0.75 seconds. I usually prefer closer to 300ms so that it feels nice and snappy.

alex-ruddell replied:
All done! :)

dikunong commented on src/components/shows/data.js:
```diff
@@ -0,0 +1,331 @@
+    "poster": revue2009,
+    "cast-picture": "null",
+    // Text
+    "title": "NZs NEXT MOST SMARTEST TOP ENGINEER WITH THE STARS",
```
Another one heh

dikunong commented on src/components/shows/showviewer.module.css:
```diff
@@ -0,0 +1,189 @@
+  vertical-align: center;
+  margin-bottom: 0px;
+  margin-top: 20px;
+  font-size: large;
```
Font sizes should ideally be set using standard elements like `<h1> <h2>` etc. This keeps font sizes consistent across pages. You should still use this class to apply any extra formatting like italics though.

alex-ruddell commented:
Hi! Have updated the things requested bar a few! A list of what I have done since it has been a fair bit of time...

- Fixed apostrophes!
- Removed "Auckland" from the title!
- Fixed font sizings
- Added a button to navigate back to show list in all cases
- _Did not store users history in "back" browser_
- _Didn't add any titles underneath posters_
- Changed hover transition
- Turned show navigation buttons into <buttons>
- Sign up button no longer breaks
- At small page widths, changed flex display to column for full page
- Added "previous show" and "next show" buttons at small widths so buttons dont budge up
- Centered posters on display so no awkward gap
- _Didn't fix the dead space, but it is centered, so should be easier to click on poster_

Hopefully everything is all good! I have had a play around with the chrome developer tool, and it proved very helpful. Also fixed some minor bugs I found along the way :)
